Event/EventBase.gd

```diff
@@ -0,0 +1,18 @@
+extends Resource
+class_name EventBase
+
+var EVENTS_CLASSIFICATION: Array[Resource] = [EventMob, EventRandom, EventShop, EventHeal]
```

Comment:
What is the purpose of events_classification ? Will every event have these 4 (and some will be empty) or are these the types that are possible ? It feels a little bit out of place to me but it might be because I'm not understanding the purpose properly

Reply:
You should read https://github.com/Saplings-Projects/1M_sub/issues/20#issuecomment-1858996318 and https://github.com/Saplings-Projects/1M_sub/issues/20#issuecomment-1861759738

Reply:
If everything is clear for you after reading those comments, merge the pull request (since you are the last one to approve)
if not, ask for me explanations on points you don't get

Reply:
Looking at the 2nd comment I'd say I'm still not 100% clear on it, reading the 1st one made me think the same thing as what Ty commented there - that it's pretty much corresponding to the light level and progression of it. 

That being said I imagine this will probably change altogether once we actually implement the light system in it's entirety ?

Reply:
Will make a drawing once i'm free today

Reply:
Tell me if you understand with this @Tomzkk 

![light_mechanic_explanation](https://github.com/Saplings-Projects/1M_sub/assets/49365680/fb0101c7-30d4-48c7-8a02-52c5de74e2f7)

Reply:
https://media.tenor.com/qhcQRwZKtJMAAAAM/it-makes-so-much-sense-jimmy-o-yang.gif

Reply:
Looks okay, I think I get it. Sounds like I had the right idea just it's not happening for all the rooms